components/sidebar/sidebar_category_list/sidebar_category_list.tsx

```diff
@@ -0,0 +1,390 @@
+    unreadChannelIds: string[];
+    isUnreadFilterEnabled: boolean;
+    handleOpenMoreDirectChannelsModal: (e: Event) => void;
+    getChannelsForCategory: (category: ChannelCategory) => Channel[];
+    actions: {
+        switchToChannelById: (channelId: string) => void;
+        close: () => void;
+    };
+};
+
+type State = {
+    showTopUnread: boolean;
+    showBottomUnread: boolean;
+};
+
+// scrollMargin is the margin at the edge of the channel list that we leave when scrolling to a channel.
+const scrollMargin = 10;
+
+// categoryHeaderHeight is the height of the category header
+const categoryHeaderHeight = 32;
+
+// scrollMarginWithUnread is the margin that we leave at the edge of the channel list when scrolling to a channel so
+// that the channel is not under the unread indicator.
+const scrollMarginWithUnread = 55;
+
+export default class SidebarCategoryList extends React.PureComponent<Props, State> {
+    channelRefs: Map<string, HTMLLIElement>;
+    scrollbar: React.RefObject<Scrollbars>;
+    animate: SpringSystem;
+    scrollAnimation: Spring;
+    closedDirectChannel: boolean;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.channelRefs = new Map();
+        this.state = {
+            showTopUnread: false,
+            showBottomUnread: false,
+        };
+        this.scrollbar = React.createRef();
+        this.closedDirectChannel = false;
+
+        this.animate = new SpringSystem();
+        this.scrollAnimation = this.animate.createSpring();
+        this.scrollAnimation.setOvershootClampingEnabled(true); // disables the spring action at the end of animation
+        this.scrollAnimation.addListener({onSpringUpdate: this.handleScrollAnimationUpdate});
+    }
+
+    componentDidMount() {
+        document.addEventListener('keydown', this.navigateChannelShortcut);
+        document.addEventListener('keydown', this.navigateUnreadChannelShortcut);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.navigateChannelShortcut);
+        document.removeEventListener('keydown', this.navigateUnreadChannelShortcut);
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (!this.props.currentChannel || !prevProps.currentChannel) {
+            return;
+        }
+
+        // reset the scrollbar upon switching teams
+        if (this.props.currentTeam !== prevProps.currentTeam) {
+            this.scrollbar.current!.scrollToTop();
+        }
+
+        // Scroll to selected channel so it's in view
+        if (this.props.currentChannel?.id !== prevProps.currentChannel?.id) { //eslint-disable-line no-undef
+            this.scrollToChannel(this.props.currentChannel!.id);
+        }
+
+        // TODO: Copying over so it doesn't get lost, but we don't have a design for the sidebar on mobile yet
+        // close the LHS on mobile when you change channels
+        if (this.props.currentChannel.id !== prevProps.currentChannel.id) {
```

Comment:
Does this also need to be `this.props.currentChannel?.id`?

Reply:
`currentChannel` isn't nullable, so really neither should have the `?` operator. I'll fix the other one.